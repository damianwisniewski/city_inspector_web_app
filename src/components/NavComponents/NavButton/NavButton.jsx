import React from 'react'

import './NavButton.scss'

import Button from '../../CommonComponents/Button/Button'

const NavButton = ({ action, ...props }) => (
	<li className='nav-button'>
		<Button {...props} onClick={action} />
	</li>
)

export default NavButton
