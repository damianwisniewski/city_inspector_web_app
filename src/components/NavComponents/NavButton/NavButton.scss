@import '../../../assets/styleHelpers/index';

$buttonColors: (
	red: $red,
	green: $green,
	white: $white,
	blue: $blue,
);

.nav-button-wrapper {
	display: flex;
	align-items: center;

	& .nav-button {
		width: 100%;
		margin: 0 10px;

		padding: 10px 20px;

		font-weight: 900;
		font-size: 14px;
		font-family: $secondary_font;

		border: none;
		border-radius: 5px;
		outline: none;

		box-shadow: $small_shadow;

		cursor: pointer;

		transition: background-color 0.1s ease-out;

		@include tablet-size {
			margin-top: 10px;
			margin-bottom: 10px;

			box-shadow: none;
		}

		// Creates separate styles for each button color
		@each $name, $color in $buttonColors {
			$font-color: $white;

			@if $name == white {
				$font-color: $dark_gray;
			}

			&.#{$name} {
				color: $font-color;

				background-color: $color;

				&:active {
					background-color: darken($color: $color, $amount: 10%);
				}
			}
		}
	}
}
