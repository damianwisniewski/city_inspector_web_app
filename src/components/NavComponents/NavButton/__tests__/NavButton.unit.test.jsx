import React from 'react'
import NavButton from '../NavButton'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('<NavButton />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let NavButtonComponent
	const ButtonText = 'ExampleText'
	const ButtonAction = jest.fn()

	beforeEach(() => {
		NavButtonComponent = shallow(<NavButton action={ButtonAction}>{ButtonText}</NavButton>)
	})

	test('should contains children passed inside', () => {
		expect(NavButtonComponent.contains(ButtonText)).toBe(true)
	})

	test('should have assigned to "onClick" prop, value of action prop', () => {
		expect(NavButtonComponent.find('Button').props().onClick).toEqual(ButtonAction)
	})

	test('should contains proper structure <li><Button/></li>', () => {
		expect(NavButtonComponent.exists('li')).toBe(true)
		expect(NavButtonComponent.find('li').exists('Button')).toBe(true)
	})
})
