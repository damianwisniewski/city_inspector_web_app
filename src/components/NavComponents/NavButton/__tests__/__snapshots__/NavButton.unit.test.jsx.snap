// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavButton /> should render properly, with props like color and to 1`] = `
<li
  className="nav-button-wrapper"
>
  <ModalLink
    className="nav-button blue"
    to="MockModal"
  >
    Example Text
  </ModalLink>
</li>
`;
